openff/qcsubmit/factories.py

```diff
@@ -691,186 +587,118 @@ class TorsiondriveDatasetFactory(OptimizationDatasetFactory):
 
-    def create_dataset(
-        self,
-        dataset_name: str,
-        molecules: Union[str, List[off.Molecule], off.Molecule],
-        description: str,
-        tagline: str,
-        metadata: Optional[Metadata] = None,
-        processors: Optional[int] = None,
-        verbose: bool = True,
-    ) -> TorsiondriveDataset:
-        """
-        Process the input molecules through the given workflow then create and populate the torsiondrive
-        dataset class which acts as a local representation for the collection in qcarchive and has the ability to
-        submit its self to a local or public instance.
-
-        Note:
-            The torsiondrive dataset allows for multiple starting geometries.
-            If fragmentation is used each molecule in the dataset will have the torsion indexes already set else indexes
-            are generated for each rotatable torsion in the molecule.
-
-        Important:
-            Any molecules with linear torsions identified for torsion driving will be removed and failed from the
-            workflow.
+    @classmethod
+    def _dataset_type(cls) -> Type[TorsiondriveDataset]:
+        return TorsiondriveDataset
 
-        Parameters:
-             dataset_name: The name that will be given to the collection on submission to an archive instance.
-             molecules: The list of molecules which should be processed by the workflow and added to the dataset, this
-                can also be a file name which is to be unpacked by the openforcefield toolkit.
-            description: A short string describing the dataset.
-            tagline: A short string overview of the collection displayed on the QCArchive.
-            metadata: Any metadata which should be associated with this dataset this can be changed from the default
-                after making the dataset.
-            processors: The number of processors avilable to the workflow, note None will use all avilable processors.
-            verbose: If True a progress bar for each workflow component will be shown.
-
-        Returns:
-            A [DataSet][qcsubmit.datasets.TorsiondriveDataset] instance populated with the molecules that have passed
-            through the workflow.
+    def _linear_torsion_filter(self, dataset: T, molecule: off.Molecule) -> None:
         """
+        Mock a linear torsion filtering component and filter the molecule.
+        """
+        try:
+            dataset.filtered_molecules["LinearTorsionRemoval"].add_molecule(
+                molecule=molecule
+            )
+        except KeyError:
+            dataset.filter_molecules(
+                molecules=[
+                    molecule,
+                ],
+                component_name="LinearTorsionRemoval",
+                component_description={
+                    "component_description": "Remove any molecules with a linear torsions selected to drive."
+                },
+                component_provenance=self.provenance(),
```

Comment:
Do these need changing now (e.g. remove `component_description`)?

Reply:
Good point, yes I think these should be removed to be consistent with the new components.